internal/tableprinter/tableprinter.go

```diff
@@ -0,0 +1,43 @@
+// Package tableprinter provides behavior to write tabular data to a given
+// destination.
+package tableprinter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+)
+
+const (
+	tabwriterMinWidth = 6
+	tabwriterWidth    = 4
+	tabwriterPadding  = 3
+	tabwriterPadChar  = ' '
+	tabwriterFlags    = tabwriter.FilterHTML
+)
+
+// NewTabWriter returns a tabwriter that transforms tabbed columns into aligned
+// text.
+func NewTabWriter(output io.Writer) *tabwriter.Writer {
+	return tabwriter.NewWriter(output, tabwriterMinWidth, tabwriterWidth, tabwriterPadding, tabwriterPadChar, tabwriterFlags)
+}
+
+// PrintTwoColumnTable uses a list of two item records (rows) to write a two
+// column table with headers to a given output destination.
+func PrintTwoColumnTable(output io.Writer, headers []string, rows [][]string) {
+	w := NewTabWriter(output)
+
+	// column headers are at the top, so they are written first
+	for _, col := range headers {
+		_, _ = fmt.Fprint(w, strings.ToUpper(col), "\t")
+	}
+	_, _ = fmt.Fprintln(w)
+
+	// rows form the body of the table
+	for _, row := range rows {
+		_, _ = fmt.Fprintln(w, row[0], "\t", row[1])
```

Comment:
I suppose I could look this up, but ... how smart is this "tab writer" about column alignment? Does the tab separator ensure the columns are aligned, or does this risk misaligning the second column when the first pushes past a tab stop?

Then again, your test output below shows all space characters, which is more or less suggested by your default parameters, so maybe I'm misinterpreting the context of the `"\t"` here?

Reply:
I think it does risk pushing past the tab alignment. From my example at the top, the tab separator does not ensure they are perfectly aligned, but they will be nearly aligned which seems good enough.

```shell
OBJECT              NAMESPACE   
HogCpuInput          $.steps.hog_cpu_wf.execute.inputs.items
```